whirlpool/appliance.py

```diff
@@ -43,8 +48,8 @@ def _event_socket_handler(self, msg):
                 continue
             self._set_attribute(attr, str(val), timestamp)
 
-        if self._attr_changed:
-            self._attr_changed()
+        for callback in self._attr_changed:
+            callback[0]()
```

Comment:
```suggestion
            callback()
```
I guess this was because you were adding the a list for each callback?